src/features/Main/index.jsx

```diff
@@ -81,29 +82,34 @@ const Main = () => {
       <Header />
       <div className="pageWrapper">
         <main className="d-flex">
-          <Sidebar />
-          <Container className="px-0 container-pages">
-            <Container size="xl" className="px-4">
-              {institutions.length > 1 && (<InstitutionSelector />)}
-            </Container>
-            <Switch>
-              <Route exact path="/">
-                <Redirect to={addQueryParam('/dashboard')} />
-              </Route>
-              {routes.map(({ path, exact, component: Component }) => (
-                <Route
-                  key={path}
-                  path={path}
-                  exact={exact}
-                  render={() => (
-                    <ActiveTabUpdater path={path}>
-                      <Component />
-                    </ActiveTabUpdater>
-                  )}
-                />
-              ))}
-            </Switch>
-          </Container>
+          {institutions.length < 1 && <UnauthorizedPage />}
```

Comment:
Is not possible to do something like:


```jsx
const Main = () => {
  // Code

  // last effect
  useEffect(() => {
    if (institutions.length === 1) {
      searchParams.set(INSTITUTION_QUERY_ID, institutions[0].id);
      history.push({ search: searchParams.toString() });
      dispatch(updateSelectedInstitution(institutions[0]));
    }
  }, [institutions, dispatch]);
```
> [!TIP]
> Stop the operation just here instead of continue doing more stuff
```jsx
 if (institutions.length === 1) {
  return <InstitutionSelector />
  }

  const routes = [
    { path: '/dashboard', component: DashboardPage, exact: true },
    { path: '/students', component: StudentsPage, exact: true },
    { path: '/instructors', component: InstructorsPage, exact: true },
    // code
```
> [!NOTE]  
> And leave the code as before

```jsx
 return (
    <BrowserRouter basename={getConfig().INSTRUCTOR_PORTAL_PATH}>
      <Header />
      <main className="d-flex page-wrapper">
        <Sidebar />
        <Container>
          <Container size="xl" className="px-4">
            {institutions?.length > 1 && (<InstitutionSelector />)}
          </Container>
          <Switch>
            <Route exact path="/">
              <Redirect to="/dashboard" />
            </Route>
            {routes.map(({ path, exact, component: Component }) => (
              <Route
                key={path}
                path={path}
      // code
```

Reply:
@01001110J  Do you mean return the component in the useeffect?

Reply:
I understood but we don't need to just return the unauthorized page, we need the other components that wrap it, so If I added that return in the if I will only return that component